Tests/ApolloTests/Cache/StoreSubscriptionTests.swift

```diff
@@ -0,0 +1,83 @@
+import XCTest
+@testable import Apollo
+
+class StoreSubscriptionTests: XCTestCase {
+  static let defaultWaitTimeout: TimeInterval = 1
+
+  var store: ApolloStore!
+
+  override func setUpWithError() throws {
+    try super.setUpWithError()
+    store = ApolloStore()
+  }
+
+  override func tearDownWithError() throws {
+    store = nil
+    try super.tearDownWithError()
+  }
+
+  // MARK: - Tests
+
+  func testSubscriberIsNotifiedOfStoreUpdate() throws {
+    let cacheKeyChangeExpectation = XCTestExpectation(description: "Subscriber is notified of cache key change")
+    let expectedChangeKeySet: Set<String> = ["QUERY_ROOT.__typename", "QUERY_ROOT.name"]
+    let subscriber = SimpleSubscriber(cacheKeyChangeExpectation, expectedChangeKeySet)
+
+    store.subscribe(subscriber)
+    addTeardownBlock { self.store.unsubscribe(subscriber) }
+
+    store.publish(
+      records: [
+        "QUERY_ROOT": [
+          "__typename": "Hero",
+          "name": "Han Solo"
+        ]
+      ]
+    )
+
+    wait(for: [cacheKeyChangeExpectation], timeout: Self.defaultWaitTimeout)
+  }
+
+  func testUnsubscribedSubscriberDoesNotReceiveStoreUpdate() throws {
```

Comment:
I know this test is correct as far as testing all the behavior thoroughly, but we try to avoid testing that something _doesn't_ happen in asynchronous context because it requires the test to run and wait for the timeout. Having multiple tests like that ends up making your test runs take too long to complete.

Instead, let's just change this to confirm that after unsubscribe is called, the subscriber is no longer in the `ApolloStore.subscribers` set. I think that should be sufficient.

You'll need to make the `subscribers` property `internal` instead of `private`. And since that is done async, the easiest way to check for it is with `Nimble`. Just add `import Nimble` to the top of this test file, then the test assertion is `expect(self.store.subscribers).toEventually(beEmpty())`.

Reply:
Makes sense! Thanks for the actionable feedback - updating now

Reply:
Updated in https://github.com/apollographql/apollo-ios-dev/pull/300/commits/6f66f7be3dcce250c43c20b11bde2b53df2b615f